### ISSUE TITLE
[A11y] Search Results - Grid - Selecting Assets

### ISSUE THREAD

**Issue by jechols** [OregonDigital/OD2#1930]
### Operating System

Windows

### Browser

Edge

### Affected page

Search

### Admin user?

Yes

### Issue Description

- "Select All" might need more context.  "Select all results on this page" maybe?
- "None" should probably be "Select None" or "Clear selected results" or something.
- Something should be announced when selections change via these buttons. Right now you hear nothing on activating the button. It should at least print out "8 items selected" or something in a hidden aria-live region.
  - It should be displayed visually, not just hidden, but I don't know if there's a WCAG criterion for this. The problem with screen-reader-only text is that if you're zoomed in a lot, it's not obvious that anything happens when you select all or none.
  - This is a usability issue as well, as it's not immediately clear I have selected anything when the search results are all images, as the checkboxes seem to be covered by the image. It would probably be really helpful to just have something indicating how many items are selected, and perhaps a clearer way of showing which items are selected.

As a side note: "Select All" and "None" are styled to look like links for some reason, which isn't a failure, but it is a bit confusing.

### Suggested Fix

This is a really tough one.  I wonder if the selection interface should only be available in "list" mode.  The same general considerations apply, but the problem of even seeing the checkbox in the first place is gone and there's more space to work with per asset.

In any case, here's my best recommendation:

- Make sure "Select All" and "None" give some feedback when activated. "13 items selected", "All items unselected", or something similar.
  - This must be in some kind of aria-live region with politeness set to "polite". https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Live_Regions
  - Consider visual feedback here, too, as that helps all users
- Consider if it makes more sense to change the selection buttons to actually look like buttons. **Not critical**, but very helpful especially for some cognitive disorders.
- Consider better wording for "Select All" and "None"? **Not critical.**
- It might be argued that the selection "area" needs instructions explaining it. These could be visually hidden, but I feel like everybody could make use of them - it's only through experimentation that you realize what it means, or learn things like why you can't select collections. A visually hidden area that's revealed on hover/focus, or a "help" link or something?

### Can anybody QA or does this need an expert?

Needs expert

### QA Steps

This whole thing is pretty complex, and there isn't really an obvious pattern I can find for something like this, so testing is pretty subjective and depends on the remediation taken.

**Comment by jechols** [comment]
It looks like the bulk-selection buttons have better labeling, but there's still dead silence on activation. I'm going to see what I can figure out about the requirements for button activation, because I'm almost certain you can't have silence on button or link activation.

Even if the announcement is just "all items selected" / "item selection cleared", I'm pretty sure we need *something* here....

**Comment by jechols** [comment]
Sounds like this can be moved to "done". The situation isn't ideal, but it's not a hard WCAG failure.
